diff -ruN a/tools/lto/CMakeLists.txt b/tools/lto/CMakeLists.txt
--- a/tools/lto/CMakeLists.txt	2016-11-19 01:03:51.000000000 +0200
+++ b/tools/lto/CMakeLists.txt	2017-11-23 12:23:03.000000000 +0200
@@ -16,7 +16,7 @@
 
 set(LLVM_EXPORTED_SYMBOL_FILE ${CMAKE_CURRENT_SOURCE_DIR}/lto.exports)
 
-add_llvm_library(LTO SHARED ${SOURCES} DEPENDS intrinsics_gen)
+add_llvm_library(LTO ${SOURCES} DEPENDS intrinsics_gen)
 
 install(FILES ${LLVM_MAIN_INCLUDE_DIR}/llvm-c/lto.h
   DESTINATION include/llvm-c
